### PR TITLE
[WIP] All-Resolution Inference

### PR DESCRIPTION
**Pull request by bthirion**
Introduction of a cluster-level inference procedure based on the all-resolution inference procedure.

### CODE REVIEW

**Comment by jeromedockes** [comment]
would it be possible to add a reference?

File: nistats/tests/test_thresholding.py [hunk 1]
```diff
@@ -103,3 +104,27 @@ def test_map_threshold():
...
+    data[2:4, 5:7, 6:8] = 5.
+    stat_img = nib.Nifti1Image(data, np.eye(4))
+    mask_img = nib.Nifti1Image(np.ones(shape), np.eye(4))
+
+    # test 1
+    th_map = cluster_level_inference(stat_img, threshold=3 ,alpha=.05)
+    vals = th_map.get_data()
+    assert_equal(np.sum(vals > 0), 8)
+
+    # test 1
```
Comment (kchawla-pi) [inline nistats/tests/test_thresholding.py]: test 2

**Comment by kchawla-pi** [comment]
Tests and a whats_new entry.

**Comment by kchawla-pi** [comment]
IS this ready to merge once CI is green, or does any more work remain @bthirion ?

**Comment by kchawla-pi** [comment]
I accidentally overwrote the entire PR.
I'm so sorry, I'll fix this.

**Pull request closed without merging**